### PR TITLE
Feature/add redis store

### PR DESCRIPTION
**Pull request by bailantaotao**

### CODE REVIEW

File: kv/redis/redis.go [hunk 1]
```diff
@@ -230,52 +226,59 @@ func (r *Redis) DeleteTree(directory string) error {
...
-func (r *Redis) AtomicPut(key string, value []byte, previous *store.KVPair, options *store.WriteOptions) (bool, *store.KVPair, error) {
+func (r *Redis) AtomicPut(key string, value []byte, previous *store.KeyValue, options ...store.PutOption) (*store.KeyValue, error) {
 	existedOne, err := r.Get(key)
 	if err != nil && err != store.ErrKeyNotFound {
-		return false, nil, err
+		return nil, err
 	}
 	if previous == nil && existedOne != nil {
-		return false, nil, store.ErrKeyModified
+		fmt.Println("exist object: ", existedOne)
```
Comment (alanchchen) [inline kv/redis/redis.go]: We have a `Logger` interface, I suggest using that.
Under `github.com/getamis/sirius/log`.

File: kv/redis/redis.go [hunk 2]
```diff
@@ -1,63 +1,60 @@
...
-	// TODO: use *redis.ClusterClient if we support miltiple endpoints
-	client := redis.NewClient(&redis.Options{
+	return new(endpoints, options...), nil
+}
+
+func new(endpoints []string, options ...RedisOption) *Redis {
+	rOption := &redis.Options{
 		Addr:         endpoints[0],
-		Password:     options.Password, //TODO: make sure that the password can be supported in *redis.ClusterClientg
+		Password:     "", //TODO: make sure that the password can be supported in *redis.ClusterClientg
```
Comment (alanchchen) [inline kv/redis/redis.go]: typo *redis.ClusterClientg

File: kv/redis/redis.go (hunk 1 above)
Comment (bailantaotao) [reply kv/redis/redis.go]: this is redundant code for testing, i will remove

File: kv/redis/redis.go (hunk 2 above)
Comment (bailantaotao) [reply kv/redis/redis.go]: ok

**Pull request merged**